packages/preact/package.json

```diff
@@ -9,7 +9,8 @@
 	],
 	"repository": {
 		"type": "git",
-		"url": "https://github.com/preactjs/signals"
+		"//": "This is a poor workaround for github.com/npm/feedback/discussions/768. Only way to get working relative links on NPM & GitHub",
+		"url": "https://github.com/preactjs/signals/blob/HEAD"
```

Comment:
You could use `/tree/` instead of `/blob`, so it works both for the homepage and files (GitHub will redirect `/tree/` to `/blob/` when necessary).

Reply:
Interesting, indeed it does!

Reply:
Oh wait, nope. NPM will not handle that correctly.

Reply:
https://www.npmjs.com/package/@rschristian/signals

https://www.runpkg.com/?@rschristian/signals@0.0.12/package.json

Repository URL is set to `https://github.com/rschristian/signals/tree/HEAD` yet the relative link on NPM is to `https://github.com/rschristian/signals/README.md#guide--api`

Reply:
I see, npmjs.com probably does some normalization and strips `/tree/<whatever>/`.

Reply:
Seemingly yes. Keeps `/blob/<whatever>` though, which is a bit odd.

Reply:
What if you use the shorthand like `"repository": "preactjs/signals"`? I looked at some other packages like [xo](https://www.npmjs.com/package/xo) and it does not seem to have this issue.

Reply:
`xo` doesn't look to contain any relative links.

That format, at least in the past, wouldn't work. Likely still won't.